src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -933,11 +940,23 @@ private JsonRpcMessage HandleInvocationTaskResult(JToken id, Task t)
                 throw new ArgumentException(Resources.TaskNotCompleted, nameof(t));
             }
 
-            if (t.IsFaulted)
+            if (t.IsFaulted && !this.ShouldCloseStreamOnException())
             {
                 return CreateError(id, t.Exception);
             }
 
+            if (t.IsFaulted && this.ShouldCloseStreamOnException())
+            {
+                if (t.Exception is AggregateException && t.Exception.InnerException != null)
```

Comment:
`Task.Exception` is *always* an `AggregateException` (it's [actually typed as such](https://msdn.microsoft.com/en-us/library/system.threading.tasks.task.exception%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396)). So you can drop that part of the check.

Then your comment mentions you want to strip off the `TargetInvocationException` but in fact it appears you're not, since you're only removing the `AggregateException`.